src/libtiled/object.h

```diff
@@ -88,6 +88,14 @@ class TILEDSHARED_EXPORT Object
     void mergeProperties(const Properties &properties)
     { Tiled::mergeProperties(mProperties, properties); }
 
+    /**
+     * Merges \a components with the existing components.
+     *
+     * Non-existing components are added.
+     * Existing components are overwritten.
```

Comment:
This sounds to me like the whole component would be replaced, while instead they are merged. I would in this case vote for letting the code speak for itself rather than trying to describe its behavior in detail here.

Reply:
Other methods in this interface have the description so I followed along. Let me know what to do.